laghos_rom.hpp

```diff
@@ -95,78 +95,109 @@ class ROM_Sampler
 
         if (input.staticSVD)
         {
-            generator_X = new CAROM::StaticSVDBasisGenerator(tH1size, max_model_dim,
+            generator_X = new CAROM::StaticSVDBasisGenerator(
+                    GenerateStaticSVDOptions(
+                      tH1size,
+                      max_model_dim
+                    ),
                     BasisFileName(VariableName::X, window, parameterID));
-            generator_V = new CAROM::StaticSVDBasisGenerator(tH1size, max_model_dim,
+            generator_V = new CAROM::StaticSVDBasisGenerator(
+                    GenerateStaticSVDOptions(
+                      tH1size,
+                      max_model_dim
+                    ),
                     BasisFileName(VariableName::V, window, parameterID));
-            generator_E = new CAROM::StaticSVDBasisGenerator(tL2size, max_model_dim,
+            generator_E = new CAROM::StaticSVDBasisGenerator(
+                    GenerateStaticSVDOptions(
+                      tL2size,
+                      max_model_dim
+                    ),
                     BasisFileName(VariableName::E, window, parameterID));
 
             if (sampleF)
             {
-                generator_Fv = new CAROM::StaticSVDBasisGenerator(tH1size, max_model_dim,
+                generator_Fv = new CAROM::StaticSVDBasisGenerator(
+                        GenerateStaticSVDOptions(
+                          tH1size,
+                          max_model_dim
+                        ),
                         BasisFileName(VariableName::Fv, window, parameterID));
-                generator_Fe = new CAROM::StaticSVDBasisGenerator(tL2size, max_model_dim,
+                generator_Fe = new CAROM::StaticSVDBasisGenerator(
+                        GenerateStaticSVDOptions(
+                          tL2size,
+                          max_model_dim
+                        ),
                         BasisFileName(VariableName::Fe, window, parameterID));
             }
         }
         else
         {
-            generator_X = new CAROM::IncrementalSVDBasisGenerator(tH1size,
-                    model_linearity_tol,
-                    false,
-                    true,
-                    max_model_dim,
```

Comment:
@chldkdtn Do we ever expect generator_X, generator_V, generator_E, generator_Fv, and generator_Fe to have different parameter values for the SVDBasisGenerators besides the dim parameter?

Reply:
I don't think so. Also, X, V, Fv should have the same dim, as should E and Fe, so the same options could be reused.

Reply:
I agree with Dylan